File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.cpp
class Solution {
public:
    int concatenatedBinary(int n) {
        int bits = 0;
        long long ans = 0;
        for(int i=1; i<=n; i++) {
            if(!(i & i-1))
                bits++;
            ans = ((ans << bits) | i) % 1000000007;
        }
        return (int) ans;
    }
};

File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.py
class Solution:
    def concatenatedBinary(self, n: int) -> int:
        bits = 0
        ans = 0
        for i in range(1, n+1):
            if i & i-1 == 0:
                bits += 1
            ans = ((ans << bits) | i) % 1000000007
        return ans